src/osdep/osdep.c

```diff
@@ -82,7 +95,9 @@ set_nonblocking_fd(int fd)
 	 * this is correct usage of winsock
 	 * when compiling for win32 */
 	u_long mode = 1; // set non-blocking socket
-	return ioctlsocket(fd, FIONBIO, &mode);
+	if (is_socket(fd)) {
```

Comment:
As above. Without this check the elinks.exe fails right on the beginning when it tries to apply non-blocking socket settings to file socket. This has no impact on other platforms.